modbus/api/endpoint.go

```diff
@@ -0,0 +1,91 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/go-kit/kit/endpoint"
+	"github.com/mainflux/edge/modbus"
+	"github.com/mainflux/mainflux/pkg/messaging"
+)
+
+var errInvalidRequest = errors.New("request option not supported")
+
+func readWriteEndpoint(pub messaging.Publisher) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (interface{}, error) {
+		switch req := request.(type) {
+		case *readRTURequest:
+			pubReq := struct {
+				Config  modbus.RTUHandlerOptions `json:"config"`
+				Options modbus.RWOptions         `json:"options"`
+			}{
+				Config: req.Config,
+				Options: modbus.RWOptions{
+					Address:  req.Address,
+					Quantity: req.Qauntity,
+				},
+			}
+			payload, err := json.Marshal(pubReq)
+			if err != nil {
+				return nil, err
+			}
+
+			if err = pub.Publish(ctx, fmt.Sprintf("modbus.read.rtu.%s", req.Datapoint), &messaging.Message{Payload: payload}); err != nil {
+				return nil, err
+			}
+			return generalResponse{Payload: []byte("successful")}, nil
+		case *readTCPRequest:
+			pubReq := struct {
+				Config  modbus.TCPHandlerOptions `json:"config"`
+				Options modbus.RWOptions         `json:"options"`
+			}{
+				Config: req.Config,
+				Options: modbus.RWOptions{
+					Address:  req.Address,
+					Quantity: req.Qauntity,
+				},
+			}
+			payload, err := json.Marshal(pubReq)
+			if err != nil {
+				return nil, err
+			}
+
+			if err = pub.Publish(ctx, fmt.Sprintf("modbus.read.tcp.%s", req.Datapoint), &messaging.Message{Payload: payload}); err != nil {
+				return nil, err
```

Comment:
Topic is hard coded. But this also raises me a question about how the data flows from edge to Mainflux , 
So We need to dicsuss about the data flow.